srml/fees/src/lib.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017-2018 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Handles all transaction fee related operations
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[macro_use]
+extern crate srml_support as runtime_support;
+
+use runtime_support::{Parameter, dispatch::Result, StorageMap};
+use runtime_primitives::traits::{As, Member, SimpleArithmetic, ChargeBytesFee, ChargeFee,
+	TransferAsset, CheckedAdd, CheckedSub, CheckedMul, Zero};
+use system;
+
+mod mock;
+mod tests;
+
+pub trait Trait: system::Trait {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+	/// The unit for fee amount
+	type Amount: Member + Parameter + SimpleArithmetic + Default + Copy + As<u64>;
+
+	/// A function does the asset transfer between accounts
+	type TransferAsset: TransferAsset<Self::AccountId, Amount = Self::Amount>;
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		fn deposit_event<T>() = default;
+
+		fn on_finalise() {
+			let extrinsic_count = <system::Module<T>>::extrinsic_count();
+			(0..extrinsic_count).for_each(|index| {
+				// Deposit `Charged` event if some amount of fee charged.
+				let fee = Self::current_transaction_fee(index.clone());
```

Comment:
You can use `take()` which will automatically remove the value after you read it:
https://github.com/paritytech/substrate/blob/master/srml/support/src/storage/generator.rs#L84

Then you can remove the last line in this loop.